render-ws-java-client/src/main/java/org/janelia/render/client/MultiSemPreAlignClient.java

```diff
@@ -0,0 +1,164 @@
+				description = "Max iterations (default:1000)"
+		)
+		public Integer maxIterations = 1000;
+
+		@Parameter(
+				names = "--maxPlateauWidth",
+				description = "Max plateau width (default:250)"
+		)
+		public Integer maxPlateauWidth = 250;
+
+		@Parameter(
+				names = "--maxNumMatches",
+				description = "Max number of matches between mFOV layers (default:1000)"
+		)
+		public Integer maxNumMatches = 1000;
+
+		@Parameter(
+				names = "--numThreads",
+				description = "Number of threads (default:8)"
+		)
+		public Integer numThreads = 8;
+	}
+
+	public static void main(final String[] args) {
+		final ClientRunner clientRunner = new ClientRunner(args) {
+			@Override
+			public void runClient(final String[] args) throws Exception {
+
+				final Parameters parameters = new Parameters();
+				parameters.parse(args);
+
+				LOG.info("runClient: entry, parameters={}", parameters);
+
+				final MultiSemPreAlignClient client = new MultiSemPreAlignClient(parameters);
+
+				client.process();
+			}
+		};
+		clientRunner.run();
+	}
+
+
+	private final Parameters parameters;
+	private final RenderDataClient dataClient;
+
+	public MultiSemPreAlignClient(final Parameters parameters) {
+		this.parameters = parameters;
+		this.dataClient = parameters.renderWeb.getDataClient();
+	}
+
+	private void setUpTargetStack() throws IOException {
+		final StackMetaData stackMetaData = dataClient.getStackMetaData(parameters.stack);
+		dataClient.setupDerivedStack(stackMetaData, parameters.targetStack);
+	}
+
+	private void completeTargetStack() throws IOException {
+		dataClient.setStackState(parameters.targetStack, StackMetaData.StackState.COMPLETE);
+	}
+
+	private void process() throws IOException, ExecutionException, InterruptedException {
+		setUpTargetStack();
+
+		final ResolvedTileSpecsWithMatchPairs tileSpecsWithMatchPairs =
+				dataClient.getResolvedTilesWithMatchPairs(parameters.stack,
+														  null,
+														  parameters.matches.matchCollection,
+														  null,
+														  null,
+														  true);
+
+		final ResolvedTileSpecCollection rtsc = tileSpecsWithMatchPairs.getResolvedTileSpecs();
+		final List<CanvasMatches> matches = tileSpecsWithMatchPairs.getMatchPairs();
+
+		final MultiSemPreAligner<RigidModel2D> preAligner = new MultiSemPreAligner<>(
+				new RigidModel2D(),
+				parameters.maxAllowedError,
+				parameters.maxIterations,
+				parameters.maxPlateauWidth,
+				parameters.numThreads,
+				parameters.maxNumMatches
+		);
+
+		final Map<String, RigidModel2D> tileIdToModel = preAligner.preAlign(rtsc, matches);
+
+		for (final Map.Entry<String, RigidModel2D> entry : tileIdToModel.entrySet()) {
+			final String tileId = entry.getKey();
+			final TransformSpec transformSpec = SolveTools.getTransformSpec(entry.getValue());
+			rtsc.addTransformSpecToTile(tileId, transformSpec, TransformApplicationMethod.PRE_CONCATENATE_LAST);
+		}
+
+		dataClient.saveResolvedTiles(rtsc, parameters.targetStack, null );
```

Comment:
Since this is a global optimization, there is no way of doing the computation by z-layer. If it's better to split saving a stack into smaller chunks, it sounds to me that the canonical place for this is doing it in `RenderDataClient` directly.